lib/key.rb

```diff
@@ -1,17 +1,32 @@
 class Key
-  attr_reader :number
+  attr_reader :number, :key_hash
 
     def initialize(number = "12345")
       @number = number
+      @key_hash = Hash.new(0)
     end
 
   def generate_random
     random = (rand(1..100000)).to_s
-    @number = ("0" * (5-(random.length)) + random)
+    ("0" * (5-(random.length)) + random)
+  end
+
+  def new_number
+    random = (rand(1..100000)).to_s
+    @number = generate_random("0" * (5-(random.length)) + random)
   end
 
   def slice_num
-    @number.to_s.split('')
+    @number.split('')
+  end
+
+  def make_key_hash
+    @key_hash = {
```

Comment:
need to make sure this is the right data type and also if abcd need to be capitalized